packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1331,7 +1331,7 @@ class SliverMultiBoxAdaptorElement extends RenderObjectElement implements Render
           indexToLayoutOffset[index] = childParentData.layoutOffset!;
         }
 
-        if (newIndex != null && newIndex != index) {
+        if (newIndex != null && newIndex != index && newIndex >= 0) {
```

Comment:
findIndexByKey probably should not return negative number at all because the build method is only called with index >= 0. we should just assert(netIndex >= 0) and add some more documentation around the findIndexByKey method

Reply:
Since `findIndex` returns -1 when not found that would make a rather annoying rough edge IMO? How about assert >= -1 to allow -1 specifically?

Reply:
the documentation said it should return null if not found

Reply:
Sure, I'm not debating what the documentation says, I'm saying it's odd that it doesn't match a common implementation, especially since this isn't caught at compile time.